Index: build/generator/vcnet_vcproj.ezt
===================================================================
--- build/generator/vcnet_vcproj.ezt	(revision 1130905)
+++ build/generator/vcnet_vcproj.ezt	(working copy)
@@ -11,6 +11,7 @@
 [for platforms][for configs]		<Configuration
 			Name="[configs.name]|[platforms]"
 			OutputDirectory="..\..\..\[configs.name]\[target.output_dir]"
+			BuildLogFile="$(IntDir)\BuildLog_$(ProjectName).htm"
 			IntermediateDirectory="..\..\..\[configs.name]\[target.intermediate_dir]"
 			ConfigurationType="[target_type]"[is configs.name "Release"]
 			WholeProgramOptimization="FALSE"[end]>
@@ -28,13 +29,17 @@
 				PreprocessorDefinitions="[if-any instrument_apr_pools]APR_POOL_DEBUG=[instrument_apr_pools];[end][for configs.defines][configs.defines][if-index configs.defines last][else];[end][end];_CRT_SECURE_NO_WARNINGS"
 [is configs.name "Debug"]				MinimalRebuild="TRUE"
 				RuntimeLibrary="3"
+				BasicRuntimeChecks="3"
 				BufferSecurityCheck="TRUE"
 				EnableFunctionLevelLinking="TRUE"
 [else]				StringPooling="TRUE"
 				RuntimeLibrary="2"
 				BufferSecurityCheck="FALSE"
-[end]				WarningLevel="3"
+[end]				WarningLevel="4"
+				DisableSpecificWarnings="4100;4127;4204;4206;4701;4706"
 				Detect64BitPortabilityProblems="FALSE"
+				AdditionalOptions="
+				/we4002 /we4003 /we4013 /we4020 /we4022 /we4024 /we4028 /we4029 /we4030 /we4031 /we4033 /we4047 /we4089 /we4113"
 				DebugInformationFormat="3"
 				ProgramDataBaseFileName="$(IntDir)\[target.output_pdb]"
 				CompileAsManaged="0"
@@ -51,7 +56,7 @@
 [else]				LinkIncremental="1"
 [end]				AdditionalLibraryDirectories="..\..\..\db4-win32\lib;[for configs.libdirs][configs.libdirs];[end]"
 				TargetMachine="[is platforms "Win32"]1[end][is platforms "x64"]17[end]"
-				IgnoreDefaultLibraryNames="libc.lib"
+				IgnoreDefaultLibraryNames="libc.lib[is configs.name "debug"];msvcrt.lib[end]"
 [if-any def_file]				ModuleDefinitionFile="[def_file]"
 [end]				GenerateDebugInformation="TRUE"
 				ProgramDatabaseFile="$(OutDir)\$(TargetName).pdb"[is configs.name "Debug"]
@@ -64,7 +69,8 @@
 			<Tool
 				Name="VCLibrarianTool"
 				AdditionalOptions="[is platforms "win32"]/MACHINE:X86[end][is platforms "x64"]/MACHINE:X64[end]"
-				OutputFile="$(OutDir)\[target.output_name]"/>
+				OutputFile="$(OutDir)\[target.output_name]"[is configs.name "debug"]
+				IgnoreDefaultLibraryNames="msvcrt.lib"[end]/>
 [end]			<Tool
 				Name="VCMIDLTool"/>
 			<Tool
