Index: engines/e_capi.c
===================================================================
--- engines/e_capi.c	(revision 34322)
+++ engines/e_capi.c	(working copy)
@@ -1489,6 +1489,13 @@
     }
     if (ctx->store_flags & CERT_SYSTEM_STORE_LOCAL_MACHINE)
         dwFlags = CRYPT_MACHINE_KEYSET;
+    if ((ptype == PROV_RSA_FULL)
+        && !wcscmp(provname, MS_ENHANCED_PROV_W)
+        && CryptAcquireContextW(&key->hprov, contname,
+                                L"Microsoft Enhanced RSA and AES Cryptographic Provider"
+                                /* = MS_ENH_RSA_AES_PROV_W, except when using Windows SDK 6.0 when compiling for XP PRE SP1 this has the " (Prototype)" suffix :( */,
+                                PROV_RSA_AES, dwFlags))
+    {} else
     if (!CryptAcquireContextW(&key->hprov, contname, provname, ptype,
                               dwFlags)) {
         CAPIerr(CAPI_F_CAPI_GET_KEY, CAPI_R_CRYPTACQUIRECONTEXT_ERROR);
