Index: subversion/libsvn_client/merge.c
===================================================================
--- subversion/libsvn_client/merge.c	(revision 935094)
+++ subversion/libsvn_client/merge.c	(working copy)
@@ -4171,7 +4171,7 @@
       else
         {
           err = svn_client__record_wc_mergeinfo(path, mergeinfo,
-                                                adm_access, subpool);
+                                                adm_access, ctx, subpool);

           if (err && err->apr_err == SVN_ERR_ENTRY_NOT_FOUND)
             {
@@ -6708,7 +6710,7 @@
       if (child->indirect_mergeinfo)
         SVN_ERR(svn_client__record_wc_mergeinfo(child->path,
                                                 child->pre_merge_mergeinfo,
-                                                adm_access,
+                                                adm_access, merge_b->ctx,
                                                 iterpool));
       if (merge_b->implicit_src_gap)
         {
@@ -6898,7 +6900,7 @@
                                             iterpool));
               SVN_ERR(svn_client__record_wc_mergeinfo(added_path,
                                                       merge_mergeinfo,
-                                                      adm_access,
+                                                      adm_access, merge_b->ctx,
                                                       iterpool));
             }
         }
@@ -7647,7 +7649,7 @@
               svn_error_t *err = svn_client__record_wc_mergeinfo(elt.key,
                                                                  elt.value,
                                                                  adm_access,
-                                                                 pool);
+                                                                 ctx, pool);

               if (err && err->apr_err == SVN_ERR_ENTRY_NOT_FOUND)
                 {
