Index: src/engine/SCons/Tool/MSCommon/common.py
===================================================================
--- src/engine/SCons/Tool/MSCommon/common.py	(revision 5600)
+++ src/engine/SCons/Tool/MSCommon/common.py	(working copy)
@@ -156,10 +156,12 @@ def get_output(vcbat, args = None, env = None):
     vs_vc_vars = [
         'COMSPEC',
         # VS100 and VS110: Still set, but modern MSVC setup scripts will
         # discard these if registry has values.  However Intel compiler setup
         # script still requires these as of 2013/2014.
+        'VS160COMNTOOLS',
+        'VS150COMNTOOLS',
         'VS140COMNTOOLS',
         'VS120COMNTOOLS',
         'VS110COMNTOOLS',
         'VS100COMNTOOLS',
         'VS90COMNTOOLS',
Index: src/engine/SCons/Tool/MSCommon/vc.py
===================================================================
--- src/engine/SCons/Tool/MSCommon/vc.py	(revision 5600)
+++ src/engine/SCons/Tool/MSCommon/vc.py	(working copy)
@@ -186,13 +186,15 @@ def get_host_target(env):
 
     return (host, target,req_target_platform)
 
 # If you update this, update SupportedVSList in Tool/MSCommon/vs.py, and the
 # MSVC_VERSION documentation in Tool/msvc.xml.
-_VCVER = ["14.1", "14.0", "14.0Exp", "12.0", "12.0Exp", "11.0", "11.0Exp", "10.0", "10.0Exp", "9.0", "9.0Exp","8.0", "8.0Exp","7.1", "7.0", "6.0"]
+_VCVER = ["14.2", "14.1", "14.0", "14.0Exp", "12.0", "12.0Exp", "11.0", "11.0Exp", "10.0", "10.0Exp", "9.0", "9.0Exp","8.0", "8.0Exp","7.1", "7.0", "6.0"]
 
 _VCVER_TO_PRODUCT_DIR = {
+    '14.2' : [
+        (SCons.Util.HKEY_LOCAL_MACHINE, r'')], # Visual Studio 2019 doesn't set this registry key anymore
     '14.1' : [
         (SCons.Util.HKEY_LOCAL_MACHINE, r'')], # Visual Studio 2017 doesn't set this registry key anymore
     '14.0' : [
         (SCons.Util.HKEY_LOCAL_MACHINE, r'Microsoft\VisualStudio\14.0\Setup\VC\ProductDir')],
     '14.0Exp' : [
Index: src/engine/SCons/Tool/MSCommon/vs.py
===================================================================
--- src/engine/SCons/Tool/MSCommon/vs.py	(revision 5600)
+++ src/engine/SCons/Tool/MSCommon/vs.py	(working copy)
@@ -196,10 +196,21 @@ class VisualStudio(object):
 #
 # If you update this list, update _VCVER and _VCVER_TO_PRODUCT_DIR in
 # Tool/MSCommon/vc.py, and the MSVC_VERSION documentation in Tool/msvc.xml.
 
 SupportedVSList = [
+    # Visual Studio 2019
+    VisualStudio('14.2',
+                 vc_version='14.2',
+                 sdk_version='10.0A',
+                 hkeys=[],
+                 common_tools_var='VS160COMNTOOLS',
+                 executable_path=r'Common7\IDE\devenv.com',
+                 batch_file_path=r'VC\Auxiliary\Build\vcvars32.bat',
+                 supported_arch=['x86', 'amd64', "arm"],
+                 ),
+
     # Visual Studio 2017
     VisualStudio('14.1',
                  vc_version='14.1',
                  sdk_version='10.0A',
                  hkeys=[],
