Index: server/request.c
===================================================================
--- server/request.c	(revision 1338359)
+++ server/request.c	(working copy)
@@ -999,7 +999,7 @@
              * case/resolving aliases.
              */
             rv = apr_stat(&thisinfo, r->filename,
-                          APR_FINFO_MIN | APR_FINFO_NAME | APR_FINFO_LINK,
+                          APR_FINFO_MIN /*| APR_FINFO_NAME*/ | APR_FINFO_LINK,
                           r->pool);
 
             if (APR_STATUS_IS_ENOENT(rv)) {
